src/ad-engine/models/ad-slot.ts

```
@@ -224,8 +224,15 @@ export class AdSlot extends EventEmitter {
 		return JSON.parse(JSON.stringify(this.config));
 	}
 
+	/**
+	 * Returns offset of slot from top of the page.
+	 *
+	 * Returns -1 if slot has no element.
+	 */
 	getTopOffset(): number {
-		return getTopOffset(this.getElement());
+		const element = this.getElement();
+
+		return element ? getTopOffset(element) : -1;
```

`-1` value might be misleading in case of using this method somewhere in the slot-related-logic (like getting top offset for viewport observer). Discussed on slack: we decided that it would be better to throw an exception here and handle it properly in the slot-tracker.